File: pyg_parent/pyg_web_manager/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../goods/findAll.action');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../goods/findPage.action?pageNum='+page+'&pageSize='+rows);
	}
	//全部商品分页
    this.findPageAll = function(page,rows){
        return $http.get("../goods/findGoodsList.action?pageNum="+page+"&pageSize="+rows);
    }
	//查询实体
	this.findOne=function(id){
		return $http.get('../goods/findOne.action?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../goods/add.action',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../goods/update.action',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../goods/delete.action?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../goods/searchForManager.action?pageNum='+page+"&pageSize="+rows, searchEntity);
	}    
	
	this.updateStatus = function(ids,status){
		return $http.get('../goods/updateStatus.action?ids='+ids+"&status="+status);
	}
});

File: pyg_parent/pyg_interface/src/main/java/com/itheima/pyg/service/goods/GoodsService.java
package com.itheima.pyg.service.goods;

import com.itheima.pyg.entity.PageResult;
import com.itheima.pyg.entity.Result;
import com.itheima.pyg.entity.vo.GoodsVo;
import com.itheima.pyg.pojo.good.Goods;
import com.itheima.pyg.pojo.item.Item;
import com.itheima.pyg.pojo.order.Order;

import java.util.List;

public interface GoodsService {
    /**
     * 运营商后台 查询全部商品,用于导出excel
     * @return
     */
    List<Goods> getGoodList();

    /**
     * 商品录入
     * @param goodsVo
     * @return
     */
    void add(GoodsVo goodsVo);

    /**
     * 分页查询商品列表
     * @param pageNum
     * @param pageSize
     * @param goods
     * @return
     */
    PageResult search(Integer pageNum, Integer pageSize, Goods goods);

    /**
     * 商品数据回显
     * @param id
     * @return
     */
    GoodsVo findOne(long id);

    /**
     * 更新商品数据
     * @param goodsVo
     */
    void update(GoodsVo goodsVo);

    /**
     * 运营商查询全部待审核商品
     * @param pageNum
     * @param pageSize
     * @param goods
     * @return
     */
    PageResult searchForManager(Integer pageNum, Integer pageSize, Goods goods);

    /**
     * 运营商进行商品审核,即修改商品状态
     * @param ids
     * @param status
     */
    void updateStatus(long[] ids, String status);

    /**
     * 逻辑删除待审核商品
     * @param ids
     */
    void delete(long[] ids);

    /**
     * 分页获取商品列表
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult<Goods> getGoodsListByPage(Integer pageNum, Integer pageSize);

    /**
     * 根据商品id查询库存列表
     * @return
     */
    List<Item> findItemList(long[] ids);

    List<String> findItemIds(long[] ids);
}
